src/shared_modules/utils/windowsHelper.h

```diff
@@ -141,6 +161,72 @@ namespace Utils
         return ret;
     }
 
+    static std::string normalizeTimestamp(const std::string& timestamp1, const std::string& timestamp2)
+    {
+        std::string ret = UNKNOWN_VALUE;
+
+        if (timestamp1.size() == INSTALLDATE_REGISTRY_VALUE_SIZE)
+        {
+            if (isNumber(timestamp1))
+            {
+                size_t pos = timestamp2.find(" ");
+
+                if (pos != std::string::npos)
+                {
+                    std::string date2_trimmed = timestamp2.substr(0, pos);
+                    std::string time2_trimmed = timestamp2.substr(pos + 1);
+
+                    Utils::replaceAll(date2_trimmed, "/", "");
+                    Utils::replaceAll(time2_trimmed, ":", "");
+
+                    if (date2_trimmed.size() == INSTALLDATE_REGISTRY_VALUE_SIZE
+                            || time2_trimmed.size() == HOURMINSEC_VALUE_SIZE)
+                    {
+                        if (date2_trimmed.compare(timestamp1) == 0)
+                        {
+                            ret = timestamp2;
+                        }
+                        else
+                        {
+                            tm* local_time_s = new tm();
```

Comment:
We can use a smart pointer. This will avoid the use of a delete and the possible memory leak if we forget the delete operation.

```suggestion
                            std::unique_ptr<tm> local_time_s { std::make_unique<tm>() };
```